pkg/ingest/conn.go

```diff
@@ -172,3 +175,39 @@ func (m *connectionManager) isEmpty() bool {
 	defer m.mtx.RUnlock()
 	return len(m.active) <= 0
 }
+
+// streamClock is a logical clock that is initialized to start at a random offset
+// that identifies a stream.
+// It always holds the next clock tick and is incremented when Read is called.
+// It's guaranteed that the 4 high bytes are 0 initially. If they overflow, the random
+// component is incremented by 1, effectively creating a new stream identifier.
+// It is not safe to be used concurrently.
+type streamClock [10]byte
```

Comment:
Is there a way to detect the same overflows with a simpler uint64?

Reply:
You mean just making it a `uint64` type to begin with? That's more or less how the implementation treats it anyway I suppose. So we could probably move to

```go
struct {
    prefix [2]byte
    clock  uint64
}
```

Code complexity wise it would just shuffle things around a bit I think.

Reply:
Appended alternate implementation. Looks a bit simpler overall.